src/sknnr_spatial/image/_dask_backed.py

```diff
@@ -52,7 +53,10 @@ def predict(
         output_dtype = ESTIMATOR_OUTPUT_DTYPES.get(estimator_type, np.float64)
 
         y_pred = da.apply_gufunc(
-            estimator._wrapped.predict,
+            # If the wrapped estimator was fit with a dataframe, it will warn about
+            # missing feature names because this passes unnamed arrays. Suppress that
+            # and let the wrapper handle feature name checks.
+            suppress_feature_name_warnings(estimator._wrapped.predict),
```

Comment:
The placement of the comment here and on lines 92-95 caught me a bit off guard.  One, because it's the identical comment and presumably it might be repeated for other functions (e.g. `transform`).  You mention this in the PR description.  Two, because the comment itself is within the `apply_gufunc` function call, it might be read as if it applies to all arguments.  Could there be a private static method in `DaskBackedWrapper` like:
```
class DaskBackedWrapper(ImageWrapper[DaskBackedType]):

    @staticmethod
    def _suppressed_estimator_function(func):
        # If the wrapped estimator was fit with a dataframe, it will warn about
        # missing feature names because this passes unnamed arrays. Suppress that
        # and let the wrapper handle feature name checks.
        return suppress_feature_name_warnings(func)

    def predict(self, ...):
        ...
        y_pred = da.apply_gufunc(self._suppressed_estimator_function(estimator._wrapped.predict, ...)
```
You were suggesting abstracting out the repeated code, so perhaps you're already ahead of me on this one.

Reply:
Agreed, it felt weird to put the same comment in both places! I like the static method suggestion, but saw an opportunity to reduce a little more duplication by refactoring out the shared arguments between the `apply_gufunc` calls into a private `_apply_gufunc` method, which also allowed for only suppressing warnings in one spot.

Let me know what you think of that choice. Also, maybe there's a more specific name than `_apply_gufunc`? I thought about `_apply_estimator_gufunc` or `_apply_sklearn_gufunc` or something similar, but I guess there's no reason this wouldn't work with other functions.

Reply:
> Agreed, it felt weird to put the same comment in both places! I like the static method suggestion, but saw an opportunity to reduce a little more duplication by refactoring out the shared arguments between the `apply_gufunc` calls into a private `_apply_gufunc` method, which also allowed for only suppressing warnings in one spot.

As always, yours is the better solution 😉.  This looks good to me and, as you say, takes out a bit more duplication.  

> Also, maybe there's a more specific name than `_apply_gufunc`? I thought about `_apply_estimator_gufunc` or `_apply_sklearn_gufunc` or something similar, but I guess there's no reason this wouldn't work with other functions.

The current name seems good to me.  It's clear that it's just an enhancement of `da.apply_gufunc` and should be able to be used with other functions as well.  I like keeping it as a more generic name.  Out of curiosity, I assume much of this code goes away if you move to `xarray.apply_ufunc` (in a future PR), so this seems like reasonable naming for the short term at least.